Gas management and transaction building integration tests

JoaoVasques opened the pull request:
gas management integration test

FelixFan1992 commented on sui.nix:
```diff
@@ -9,13 +9,13 @@ stdenv.mkDerivation rec {
     pkgs.fetchzip {
       url = "https://github.com/MystenLabs/sui/releases/download/mainnet-v${version}/sui-mainnet-v${version}-macos-arm64.tgz"; # Assume is a M1 Mac
-      sha256 = "sha256-JRg6hVVJed4LhH5OhQggSbOBZjZiUSuGCK5lHGtU49g=";  # Should be replaced when bumping versions
+      sha256 = "sha256-XyViJhgf6ZG1e2Y8sJp60h80O8ufLf2hmC2ZjnZadxg=";  # Should be replaced when bumping versions
```
should not update these when no SUI version upgrades?

JoaoVasques replied:
You're right, no idea how this thing went through 😢

JoaoVasques replied:
@FelixFan1992 reverted

JoaoVasques committed:
Update transacton generation

- Add transaction generation integration test
- Add selected coins as part of SuiTx struct
- Fix coin selection logic

JoaoVasques committed:
Merge branch 'develop' into gas-management-integration-test

JoaoVasques committed:
Ensure no parallel test

JoaoVasques committed:
Merge branch 'develop' into gas-management-integration-test

JoaoVasques committed:
Merge branch 'develop' into gas-management-integration-test

JoaoVasques committed:
Add retry logic to faucet request

JoaoVasques commented:
@FelixFan1992 Can you give another ✅ please?

FelixFan1992 committed:
Merge branch 'develop' into gas-management-integration-test
